File: backend/model/article.go
package model

import "github.com/google/uuid"

type Article struct {
	Id      uuid.UUID `json:"id"`
	Title   string    `json:"title"`
	Content string    `json:"content"`
	Date    string    `json:"date"`
}

type ArticlePreviewDTO struct {
	Blogs []struct {
		Id    uuid.UUID `json:"id"`
		Title string    `json:"title"`
	} `json:"blogs"`
}

File: frontend/lib/model/article.dart
import 'package:json_annotation/json_annotation.dart';

part 'article.g.dart';

@JsonSerializable()
class Article {
  final String id;
  final String title;
  final String content;
  final String date;

  Article({
    required this.id,
    required this.title,
    required this.content,
    required this.date,
  });

  factory Article.fromJson(Map<String, dynamic> json) =>
      _$ArticleFromJson(json);

  Map<String, dynamic> toJson() => _$ArticleToJson(this);
}

List<Article> articleList = [];
